vertical-pod-autoscaler/hack/generate-crd-yaml.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPOSITORY_ROOT=$(realpath $(dirname ${BASH_SOURCE})/..)
+CRD_OPTS=crd:trivialVersions=false,allowDangerousTypes=true
+APIS_PATH=${REPOSITORY_ROOT}/pkg/apis
+OUTPUT=${REPOSITORY_ROOT}/deploy/vpa-v1-crd-v1.yaml
+WORKSPACE=$(mktemp -d)
+
+function cleanup() {
+    rm -r ${WORKSPACE}
+}
+trap cleanup EXIT
+
+if [[ -z $(which controller-gen) ]]; then
+    (
+        cd $WORKSPACE
+	      go mod init tmp
+	      go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.0
+    )
+    CONTROLLER_GEN=${GOBIN:-$(go env GOPATH)/bin}/controller-gen
+else
+    CONTROLLER_GEN=$(which controller-gen)
+fi
+
+# The following commands always returns an error because controller-gen does not accept keys other than strings.
+${CONTROLLER_GEN} ${CRD_OPTS} paths="${APIS_PATH}/..." output:crd:dir=${WORKSPACE} ||:
```

Comment:
Currently, controller-gen does not support feature to check for specific error or ignore only the specific error, but we can find other errors in the output.  If the generator fails for a different error, this script will also output other than `map keys must be strings`.

```
$ hack/generate-crd-yaml.sh
/Users/ladicle/src/k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1/types.go:360:20: map keys must be strings, not int
/Users/ladicle/src/k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1beta2/types.go:340:20: map keys must be strings, not int
Error: not all generators ran successfully
run `controller-gen crd:trivialVersions=false,allowDangerousTypes=true paths=/Users/ladicle/src/k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/... output:crd:dir=/var/folders/p_/hzm_gclx7vs5cdbfbzpl0s4w0000gn/T/tmp.Cytp8OtyCn -w` to see all available markers, or `controller-gen crd:trivialVersions=false,allowDangerousTypes=true paths=/Users/ladicle/src/k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/... output:crd:dir=/var/folders/p_/hzm_gclx7vs5cdbfbzpl0s4w0000gn/T/tmp.Cytp8OtyCn -h` for usage
```

Another way is to grep the controller-gen output and display a warning message.

Reply:
If there is a not-overly complicated way to indicate that something other went wrong, I would really like to see it. Otherwise it might get really confusing for someone encountering issues.